R/plotImage.R

```diff
@@ -31,6 +33,89 @@ NULL
 #' @export
 plotSpatialData <- \() ggplot() + scale_y_reverse() + .theme 
 
+# merge/manage image channels
+# if no colors and channels defined, return the first channel
+#' @noRd
+.manage_channels <- \(a, ch, c=NULL){
+  if(length(ch) > length(.DEFAULT_COLORS) && is.null(c))
+    stop("You can only choose at most seven default colors!")
+  if(!is.null(c) || (is.null(c) && length(ch) > 1)) {
+    if(is.null(c))
+      c <- .DEFAULT_COLORS[1:length(ch)] 
+    c <- col2rgb(c)/255
+    a_new <- array(0, dim = c(3,dim(a)[-1]))
+    for(i in 1:dim(a)[1]){
+      a_new[1,,] <-  a_new[1,,,drop = FALSE] + a[i,,,drop = FALSE]*c[1,i]
+      a_new[2,,] <-  a_new[2,,,drop = FALSE] + a[i,,,drop = FALSE]*c[2,i]
+      a_new[3,,] <-  a_new[3,,,drop = FALSE] + a[i,,,drop = FALSE]*c[3,i]
+    }
+    a <- pmin(a_new,1)
+  } else {
+    a <- a[rep(1,3),,]
+  }
+  a
+}
+
+# check if an image is rgb or not
+#' @importFrom SpatialData getZarrArrayPath
+#' @importFrom Rarr zarr_overview
+#' @noRd
+.get_image_dtype <- \(a){
+  zarray_spec <- Rarr::zarr_overview(getZarrArrayPath(a), 
+                                     as_data_frame = TRUE)
+  if("data_type" %in% names(zarray_spec))
+    return(zarray_spec$data_type)
+  return(NULL)
+}
+
+# normalize the image data given its data type
+#' @noRd
+.normalize_image_array <- \(a, dt){
+  if(dt %in% names(.DTYPE_MAX_VALUES)) a <- a/.DTYPE_MAX_VALUES[[dt]]
```

Comment:
can u clarify - are there missing curly brackets here? looks like the `else` will be evaluated in any case, which will give an error?